From a2ad800c08c580dc21d994a6a932a9c7d05c2da2 Mon Sep 17 00:00:00 2001
From: Wenping Zhang <wenping.zhang@rock-chips.com>
Date: Tue, 5 Dec 2017 09:01:42 +0800
Subject: [PATCH] leds: leds-is31fl32xx: fix compile errors.

The is32fl32xx led driver from upstream is written for new led
framework, so just modify this driver for some api difference
between our current led framework.

Change-Id: I0fd9af4bc2cd419d3e0bcd1b2348d34d166d652b
Signed-off-by: Wenping Zhang <wenping.zhang@rock-chips.com>
---
 drivers/leds/leds-is31fl32xx.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/drivers/leds/leds-is31fl32xx.c b/drivers/leds/leds-is31fl32xx.c
index 31a9d749c8be..b2e7dbe7d7ad 100644
--- a/drivers/leds/leds-is31fl32xx.c
+++ b/drivers/leds/leds-is31fl32xx.c
@@ -223,7 +223,7 @@ static int is31fl3216_software_shutdown(struct is31fl32xx_priv *priv,
  * are equivalent. Poking the Update register merely applies all PWM
  * register writes up to that point.
  */
-static int is31fl32xx_brightness_set(struct led_classdev *led_cdev,
+static void is31fl32xx_brightness_set(struct led_classdev *led_cdev,
 				     enum led_brightness brightness)
 {
 	const struct is31fl32xx_led_data *led_data =
@@ -244,9 +244,11 @@ static int is31fl32xx_brightness_set(struct led_classdev *led_cdev,
 			       cdef->pwm_register_base + pwm_register_offset,
 			       brightness);
 	if (ret)
-		return ret;
+		dev_err(led_cdev->dev,
+			"set brightness %d for led[%d] failed\n",
+			brightness, pwm_register_offset);
 
-	return is31fl32xx_write(led_data->priv, cdef->pwm_update_reg, 0);
+	is31fl32xx_write(led_data->priv, cdef->pwm_update_reg, 0);
 }
 
 static int is31fl32xx_reset_regs(struct is31fl32xx_priv *priv)
@@ -335,7 +337,7 @@ static inline size_t sizeof_is31fl32xx_priv(int num_leds)
 }
 
 static int is31fl32xx_parse_child_dt(const struct device *dev,
-				     const struct device_node *child,
+				     struct device_node *child,
 				     struct is31fl32xx_led_data *led_data)
 {
 	struct led_classdev *cdev = &led_data->cdev;
@@ -357,7 +359,7 @@ static int is31fl32xx_parse_child_dt(const struct device *dev,
 	of_property_read_string(child, "linux,default-trigger",
 				&cdev->default_trigger);
 
-	cdev->brightness_set_blocking = is31fl32xx_brightness_set;
+	cdev->brightness_set = is31fl32xx_brightness_set;
 
 	return 0;
 }
-- 
2.35.3

